prototypes/ScheduledMerges.hs

```diff
@@ -281,13 +277,15 @@ newMerge tr level mergepolicy mergelast rs = do
   where
     cost = sum (map Map.size rs)
     -- How much we need to discharge before the merge can be guaranteed
-    -- complete.
+    -- complete. More precisely, this is the maximum amount a merge at this
+    -- level could need. This overestimation means that merges will only
+    -- complete at the last possible moment.
```

Comment:
Well, that depends: the merge can finish earlier, but it should never finish too late. Maybe reword this a little bit?

Reply:
In a real implementation that's true. But in the prototype merges will never finish early, as their completion is not based on the any "real" work to be performed (which would depend on the number of entries), but on the debt calculated here, which is the maximum a merge at this level could require.

I can add that explanation to the comment, though.

Reply:
I think the reasoning for always requiring the maximum amount of work is that it makes it more likely to find issues related to supplying too few credits somewhere. We could base everything on `cost` (sum of input run sizes), but not sure whether it's an improvement or not.